src/inmanta/compiler/__init__.py

```diff
@@ -213,11 +213,14 @@ def compile(self) -> Tuple[List["Statement"], List["BasicBlock"]]:
                 ns = ns.get_child(part)
 
             if ns is None:
-                raise Exception("Unable to find namespace for plugin module %s" % (cls.__module__))
+                # Un-register this plugin, as it is part of a module that is not loaded
+                # (https://github.com/inmanta/inmanta-core/issues/5651)
+                PluginMeta.clear(".".join(mod_ns))
```

Comment:
We could even further optimize this by defining a new `clear()` method that accepts a set of namespaces. Then we can do the clear after the for loop by doing a single pass over the `__functions` dictionary. That would give us O(n) performance in terms of the number of plugin functions.

Reply:
I agree. The current `clear` method must remain as is (stable API) but adding a second method to clear/unregister multiple modules sounds like a good idea. You should make sure it also works for submodules, because I think the current `clear` method does not (and is not intended to). We should probably have a test case for that scenario.

Alternatively, an even simpler option that I hadn't considered before is to just leave them in and ignore them. The important thing is we don't add the `PluginStatement`. So we could keep the implementation here mostly as is but drop the `PluginMeta.clear()` call. This may also be more reliable for in-process recompiles like pytest-inmanta because we keep the plugin objects around. And I don't immediately see any bad side effects. Wdyt?

Reply:
I like the proposal to leave them in and just ignore them.